src/ngraph/ngraph_emitter.cc

```diff
@@ -1508,6 +1511,10 @@ void Emitter::CreateLossOps() {
 
     return grad;
   };
+  loss_op_backward_funcs_["stop_gradient"] = [this](
+      const NodePtr& node, const NgraphNodePtr& adjoint) {
+    return makeConstant(node->inputs_[0], "0");
+  };
```

Comment:
I don't think this will work, the manual backward function is only applied to the last node in a subgraph.

Reply:
right,  i notice it fails below. we expect 0 for first output of grad. We could insert zero grad node for certain ops? Is there an optimal way to do this.

```
import mxnet as mx
v1 = mx.nd.array([[1, 2]])
v2 = mx.nd.array([[0, 1]])
a = mx.sym.Variable('a')
b = mx.sym.Variable('b')
b_stop_grad = mx.sym.stop_gradient(3 * b)
loss = mx.sym.MakeLoss(b_stop_grad + a)

executor = loss.simple_bind(ctx=mx.cpu(), a=(1,2), b=(1,2))
executor.forward(is_train=True, a=v1, b=v2)
print(executor.outputs)
executor.backward()
print(executor.grad_arrays)
```

Reply:
I recall running unit-tests and this passing. I guess we cant rely on it fully.

Reply:
Yeah, if stop_gradient is the last node in the graph, this is fine, but otherwise this fails. I think we need a stop_gradient op in nGraph, it's a no-op forward and doesn't have a defined generate adjoints.